cpp/src/slice2cs/CsUtil.cpp

```diff
@@ -1089,14 +1089,26 @@ Slice::CsGenerator::writeTaggedMarshalCode(Output& out,
         TypePtr keyType = d->keyType();
         TypePtr valueType = d->valueType();
 
+        bool withBitSequence = false;
+
+        if (auto optional = OptionalPtr::dynamicCast(valueType);
+            optional && !optional->underlying()->isClassType() && !optional->underlying()->isClassType())
```

Comment:
We have this in multiple places, perhaps there's a way to improve this check? Could we extend Optional for this case perhaps?

Reply:
Yes, I added a new function `encodedUsingBitSequence`.